malcolm/core/process.py

```diff
@@ -28,12 +36,20 @@ def recv_loop(self):
             if request is PROCESS_STOP:
                 # Got the sentinel, stop immediately
                 break
-            try:
-                self.handle_request(request)
-            except Exception:
-                # TODO: request.respond_with_error()
-                self.log_exception("Exception while handling %s",
-                                   request.to_dict())
+            elif isinstance(request, BlockNotify):
```

Comment:
How about using a function dictionary as a switch statement:
```python
self.handle_functions = {
    Request.POST: self._forward_block_request,
    Request.PUT: self._forward_block_request,
    Request.SUBSCRIBE: self._handle_subscribe,
    BlockNotify.type_: self._handle_block_notify
}
```
Of course you have to do BlockNotify.type_ = "BlockNotify" somewhere...
Then you can use it like:
```python
try:
    self.handle_functions[request.type_](request)
except Exception:
    self.log_exception(...)
```